src/core-sdk-tasks/BuildFPMToolPreReqs.cs

```diff
@@ -116,32 +115,29 @@ private string BuildCmdParameters(ConfigJson configJson, string package_version)
             //      --verbose : Set verbose output for FPM tool     --Static  
             //      <All folder mappings> : Add all the folder mappings for package_root, docs, man pages   --Static
 
-            var parameters = new List<string>();
-            parameters.Add("-s dir");
-            parameters.Add("-t rpm");
-            parameters.Add(string.Concat("-n ", configJson.Package_Name));
-            parameters.Add(string.Concat("-v ", package_version));
-            parameters.Add(string.Concat("-a ", configJson.Control.Architecture));
+            var parameters = new List<string>
+            {
+                "-s dir",
+                "-t rpm",
+                string.Concat("-n ", configJson.Package_Name),
+                string.Concat("-v ", package_version),
+                string.Concat("-a ", configJson.Control.Architecture)
+            };
 
             // Build the list of dependencies as -d <dep1> -d <dep2>
             if (configJson.Rpm_Dependencies != null)
             {
                 foreach (RpmDependency rpmdep in configJson.Rpm_Dependencies)
                 {
-                    string dependency = "";
                     if (rpmdep.Package_Name != "")
                     {
                         // If no version is specified then the dependency is just the package without >= check
-                        if (rpmdep.Package_Version == "")
-                        {
-                            dependency = rpmdep.Package_Name;
-                        }
-                        else
-                        {
-                            dependency = string.Concat(rpmdep.Package_Name, " >= ", rpmdep.Package_Version);
-                        }
+                        string dependency = rpmdep.Package_Version != "" ?
+                            string.Concat(rpmdep.Package_Name, " >= ", rpmdep.Package_Version) :
+                            rpmdep.Package_Name;
+
+                        parameters.Add(string.Concat("-d ", EscapeArg(dependency)));
                     }
-                    if (dependency != "") parameters.Add(string.Concat("-d ", EscapeArg(dependency)));
```

Comment:
![image](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExOHphbHNmamgwYXkxdmRqeDhvb2ZlbzI0NG03bTZrZzA3ZHZyaXI1MyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/tu54GM19sqJOw/giphy.gif)

So, if `dependency` is not empty, then add the parameter. Makes sense. If `Package_Name` is not empty, then set dependency. The if-else always sets dependency. Why? Because the truthy body sets it to `Package_Name`, which is already guaranteed to not be empty. The falsey body always sets it to a non-empty value. So, it is guaranteed to always add the parameter.

Reply:
pedantic me would say use nullorwhitespace or is empty string  signficant here?